src/cloudWatchLogs/activation.ts

```diff
@@ -66,6 +66,6 @@ export async function activate(context: vscode.ExtensionContext, configuration:
         // Here instead of in ../awsexplorer/activation due to dependence on the registry.
         Commands.register('aws.cwl.viewLogStream', async (node: LogGroupNode) => await viewLogStream(node, registry)),
 
-        Commands.register('aws.cwl.searchLogGroup', async () => await searchLogGroup(registry))
+        Commands.register('aws.cwl.searchLogGroup', async (node: LogGroupNode) => await searchLogGroup(node, registry))
```

Comment:
This change is lying to you:
```suggestion
        Commands.register('aws.cwl.searchLogGroup', async (node?: LogGroupNode) => await searchLogGroup(node, registry))
```
Lying types is a dangerous game. See `null` in older languages for a perfect example.

Reply:
I see, so with this change the type of node within `searchLogGroup` is `LogGroupNode | undefined` as well right?

Reply:
>  I see, so with this change the type of node within searchLogGroup is LogGroupNode | undefined as well right?

Yes. The interesting thing about Typescript is that it's self contained. Just because we say that a certain type is expected, that doesn't mean VS Code will give it to us. If we wanted to be _very_ strict  then we should do:
```ts
        Commands.register('aws.cwl.searchLogGroup', async (arg?: unknown) => {
            if (arg === undefined || arg instanceof LogGroupNode) {
                return searchLogGroup(arg, registry)
            } else {
                throw new TypeError('Expected argument to be `undefined` or `LogGroupNode`')
            }
        })
```

Anything that doesn't come from within the codebase should be treated as `unknown`.

Reply:
How is what you proposed stricter than having `searchLogGroup` expect a `LogGroupNode | undefined`? Wouldn't they throw the same error if the wrong type was passed in? What is the advantage of treating it as `unknown`? Is it that since its not from the codebase we don't wan't to make any assumptions?

Reply:
> How is what you proposed stricter than having searchLogGroup expect a LogGroupNode | undefined?

Typescript performs zero runtime validation of these types. VS Code is the one calling this function, not us. Anything can be passed into this function and so `LogGroupNode | undefined` is not accurate. The fact that Typescript lets you do this is because the whole point of Typescript is to give developers complete control over the types. 

You can try it for yourself:
```ts
vscode.commands.executeCommand('aws.cwl.searchLogGroup', 'hi')
```

What error do you see? Is it obvious what went wrong?

Reply:
It looks like it complains that `'hi` is not a function, which makes sense it's expecting one.

Reply:
No you should see something like this:
```
TypeError: Cannot read properties of undefined (reading 'logGroupName')
```

Reply:
Oh I did that wrong the first time. I see what you mean now. Yeah, it doesn't tell you that you broke the type contract, which isn't very helpful. It lets you pass in a type its not expecting, but then still tries to run the code with it, it seems.